pkg/apps/sonarr.go

```diff
@@ -1112,6 +1115,82 @@ func sonarrDeleteAllCustomFormats(req *http.Request) (int, interface{}) {
 	}
 }
 
+// @Description  Returns all Import Lists from Sonarr.
+// @Summary      Get Sonarr Import Lists
+// @Tags         Sonarr
+// @Produce      json
+// @Param        instance  path   int64  true  "instance ID"
+// @Success      200  {object} apps.Respond.apiResponse{message=[]sonarr.ImportListOutput}  "import list list"
+// @Failure      500  {object} apps.Respond.apiResponse{message=string} "instance error"
+// @Failure      404  {object} string "bad token or api key"
+// @Router       /api/sonarr/{instance}/importlist [get]
+// @Security     ApiKeyAuth
+func sonarrGetImportLists(req *http.Request) (int, interface{}) {
+	ilist, err := getSonarr(req).GetImportListsContext(req.Context())
+	if err != nil {
+		return apiError(http.StatusInternalServerError, "getting import lists", err)
+	}
+
+	return http.StatusOK, ilist
+}
+
+// @Description  Updates an Import List in Sonarr.
+// @Summary      Update Sonarr Import List
+// @Tags         Sonarr
+// @Produce      json
+// @Accept       json
+// @Param        instance  path   int64  true  "instance ID"
+// @Param        listID  path   int64  true  "Import List ID"
+// @Param        PUT body sonarr.ImportListInput  true  "Updated Import Listcontent"
+// @Success      200  {object} apps.Respond.apiResponse{message=sonarr.ImportListOutput}  "import list returns"
+// @Failure      400  {object} apps.Respond.apiResponse{message=string} "invalid json provided"
+// @Failure      500  {object} apps.Respond.apiResponse{message=string} "instance error"
+// @Failure      404  {object} string "bad token or api key"
+// @Router       /api/sonarr/{instance}/importlist/{listID} [put]
+// @Security     ApiKeyAuth
+func sonarrUpdateImportList(req *http.Request) (int, interface{}) {
+	var ilist sonarr.ImportListInput
+	if err := json.NewDecoder(req.Body).Decode(&ilist); err != nil {
+		return apiError(http.StatusBadRequest, "decoding payload", err)
+	}
+
+	ilist.ID, _ = strconv.ParseInt(mux.Vars(req)["ilid"], mnd.Base10, mnd.Bits64)
```

Comment:
You could read the ID from the payload instead of require it in the param. Lemme know if you want to change that.

Reply:
This was copied from one of the other existing apps so it would need to be changed on more than just this one

Reply:
I resoled the other 2, this one i'll let you take care of :)